src/status_im2/contexts/onboarding/create_password/view.cljs

```diff
@@ -1,42 +1,197 @@
+         :size  :paragraph-2}
+        text]])]])
+
+(defn password-inputs
+  [{:keys [passwords-match? on-change-password on-change-repeat-password on-input-focus
+           password-long-enough? empty-password? same-password-lenght?]}]
+  (reagent/with-let [show-hint-2? (reagent/atom false)]
+    (let [_ (when same-password-lenght? (reset! show-hint-2? true))
+          hint-1-status (if password-long-enough? :success :neutral)
+          hint-2-status (if passwords-match? :success :danger)
+          hint-2-text   (if passwords-match?
+                          (i18n/label :t/password-creation-match)
+                          (i18n/label :t/password-creation-dont-match))
+          error?        (and @show-hint-2?
+                             (not passwords-match?)
+                             (not empty-password?))]
+      [rn/view
+       [password-with-hint
+        {:hint           {:text   (i18n/label :t/password-creation-hint)
+                          :status hint-1-status
+                          :shown  true}
+         :placeholder    (i18n/label :t/password-creation-placeholder-1)
+         :on-change-text on-change-password
+         :on-focus       #(on-input-focus :password)}]
+       [rn/view {:style style/space-between-inputs}]
+       [password-with-hint
+        {:hint           {:text   hint-2-text
+                          :status hint-2-status
+                          :shown  (and (not empty-password?)
+                                       @show-hint-2?)}
+         :error?         error?
+         :placeholder    (i18n/label :t/password-creation-placeholder-2)
+         :on-change-text on-change-repeat-password
+         :on-focus       #(on-input-focus :repeat-password)
+         :on-blur        #(if empty-password?
+                            (reset! show-hint-2? false)
+                            (reset! show-hint-2? true))}]])))
+
+(def strength-status
+  {1 :very-weak
+   2 :weak
+   3 :okay
+   4 :strong
+   5 :very-strong})
+
+(defn help
+  [{{:keys [lower-case? upper-case? numbers? symbols?]} :validations
+    password-strength                                   :password-strength}]
+  [rn/view
+   [quo/strength-divider {:type (strength-status password-strength :info)}
+    (i18n/label :t/password-creation-tips-title)]
+   [rn/view {:style style/password-tips}
+    [quo/tips {:completed? lower-case?}
+     (i18n/label :t/password-creation-tips-1)]
+    [quo/tips {:completed? upper-case?}
+     (i18n/label :t/password-creation-tips-2)]
+    [quo/tips {:completed? numbers?}
+     (i18n/label :t/password-creation-tips-3)]
+    [quo/tips {:completed? symbols?}
+     (i18n/label :t/password-creation-tips-4)]]])
+
+(defn has-lower-case [s] (re-find #"[a-z]" s))
+(defn has-upper-case [s] (re-find #"[A-Z]" s))
+(defn has-numbers [s] (re-find #"\d" s))
+(defn has-symbols [s] (re-find #"[^a-zA-Z0-9\s]" s))
+(defn at-least-10-chars? [s] (>= (count s) 10))
+
+(defn password-validations
+  [s]
```

Comment:
Maybe `password` instead of `s`? In general I prefer longer but more descriptive names instead of single letter ones like `s`, but dunno if we have agreed on something on this, I remember to read a discussion about this before

Reply:
Usually in Clojure devs use `s`  to indicate the function is receiving a string :thinking:  I'm not sure then and I don't have any strong opinion on it.

Probably @ilmotta  or @flexsurfer  can help us making this clear.

Reply:
Yeah not a big deal, I know it is a common practice in Clojure but I don't like it, also I have no problem to adapt myself but would be nice to hear other opinions and reach consensus on this. I have no problem to update our guidelines afterwards so we can stick to it on following PRs :)

Reply:
i usually use `value` but s is fine if clojure devs use it, i hear that fist time

Reply:
`s` is the idiomatic choice when the author of the function wants to emphasize that the function doesn't care about the actual meaning of the string, rather only the type (string) or some interface(s) (e.g. to use `coll` to mean any of the supported Clojure collections).

So that's what I know about how other Clojure devs judge if they want to use one convention over the other.

But it's all a moot point :man_shrugging: anything will do IMO.

Reply:
I renamed the var to `password` but kept the `s` in the functions moved to `utils.string`. IMO it is clear and is consistent with the existing one there (`truncate-str-memo` receives `s`, `threshold` and opt `middle?`)
![image](https://user-images.githubusercontent.com/90291778/226731984-26a4d6be-8a26-4195-8303-38abfbe690a0.png)